datafusion/functions/src/datetime/date_part.rs

```diff
@@ -223,6 +233,30 @@ impl ScalarUDFImpl for DatePartFunc {
     }
 }
 
+fn is_integar_part(part: &str) -> bool {
+    let part = part_normalization(part);
+    matches!(
+        part.to_lowercase().as_str(),
+        "year"
+            | "month"
+            | "week"
+            | "day"
+            | "hour"
+            | "minute"
+            | "qtr"
+            | "quarter"
+            | "doy"
+            | "dow"
+    )
+}
+
+// Try to remove quote if exist, if the quote is invalid, return original string and let the downstream function handle the error
+fn part_normalization(part: &str) -> &str {
+    part.strip_prefix(|c| c == '\'' || c == '\"')
+        .and_then(|s| s.strip_suffix(|c| c == '\'' || c == '\"'))
```

Comment:
why would we want to support unmatched quotes like `'..."`?

the date_part argument shouldn't contain any quotes, why would we want to support quotes at all?


i see this is pre-existing, so okay to keep, but still wonder why we're doing this, doesn't feel right

Reply:
```
postgres=# SELECT EXTRACT(second FROM timestamp '2020-09-08T12:00:12.123456789+00');
  extract  
-----------
 12.123457
(1 row)

postgres=# SELECT EXTRACT('second' FROM timestamp '2020-09-08T12:00:12.123456789+00');
  extract  
-----------
 12.123457
(1 row)
```

It seems this it to be postgres-compatible